Code/GraphMol/Substruct/SubstructUtils.cpp

```diff
@@ -217,6 +217,14 @@ std::vector<MatchVectType> sortMatchesByDegreeOfCoreSubstitution(
   return matchScorer.sortMatchesByDegreeOfCoreSubstitution();
 }
 
+bool isAtomTerminalRGroupOrQueryHydrogen(const Atom *atom) {
+  return atom->getDegree() == 1 &&
+         (atom->getAtomicNum() == 0 ||
+          (atom->hasQuery() &&
+           describeQuery(atom).find("AtomAtomicNum 1 = val") !=
```

Comment:
using `describeQuery()` here is a nice solution to something that's otherwise a pain in the ass to deal with

Reply:
At first it seemed too easy, but I tried a few different ways to break it and couldn't ;-)